From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Wed, 17 Jun 2015 16:28:12 +0000
Subject: Ignore invalid X-Plasma-ConfigPlugins
X-Git-Url: http://quickgit.kde.org/?p=plasma-framework.git&a=commitdiff&h=ac04b5570a4072dcb0365387bc739c870a68fe65
---
Ignore invalid X-Plasma-ConfigPlugins

CHANGELOG: Fix occasional empty applet config pages

CCBUG: 349250
REVIEW: 124113
---


--- a/src/plasmaquick/configview.cpp
+++ b/src/plasmaquick/configview.cpp
@@ -145,6 +145,11 @@
             KPluginLoader loader(KPluginLoader::findPlugin(QLatin1String("kcms/") + kcm));
             KPluginMetaData md(loader.fileName());
 
+            if (!md.isValid()) {
+                qWarning() << "Could not find" << kcm << "specified in X-Plasma-ConfigPlugins";
+                continue;
+            }
+
             configModel->appendCategory(md.iconName(), md.name(), QString(), loader.fileName());
         }
     }

